server/sonar-main/src/main/java/org/sonar/application/process/ProcessLauncherImpl.java

```diff
@@ -88,6 +90,20 @@ public ProcessMonitor launch(EsCommand esCommand) {
     }
   }
 
+  private static void cleanupOutdatedEsData(EsCommand esCommand) {
+    EsFileSystem esFileSystem = esCommand.getFileSystem();
+    esFileSystem.getOutdatedDataDirectories().forEach(outdatedDir -> {
+      if (outdatedDir.exists()) {
+        LOG.info("Deleting outdated search index data directory {}", outdatedDir.getAbsolutePath());
```

Comment:
@bartfastiel there is an inconsistency here.

you added support for multiple outdated data dir (which is overkill since there is only one such directory) but the log will work only for a single such directory, always the search directory.

I'd remove the useless collection and avoid the problem

Reply:
I created PR #2597 for to avoid misunderstandings with the variable names.